File: ios/SubTrackerWidget/WidgetData.swift
import Foundation
import WidgetKit

/// Data structure matching the JSON sent from React Native
struct WidgetData: Codable {
    let currency: String
    let upcomingPayments: [UpcomingPayment]
    let translations: WidgetTranslations?
}

/// Translations for widget UI
struct WidgetTranslations: Codable {
    let upcoming: String
    let noPayments: String
    let today: String
    let dayChar: String
}

/// Upcoming payment information
struct UpcomingPayment: Codable {
    let name: String
    let amount: Double
    let daysLeft: Int
}

/// Timeline entry for the widget
struct SubTrackerEntry: TimelineEntry {
    let date: Date
    let data: WidgetData?
    
    /// Placeholder entry for widget gallery
    static var placeholder: SubTrackerEntry {
        SubTrackerEntry(
            date: Date(),
            data: WidgetData(
                currency: "$",
                upcomingPayments: [
                    UpcomingPayment(name: "Netflix", amount: 15.99, daysLeft: 2),
                    UpcomingPayment(name: "Spotify", amount: 9.99, daysLeft: 5),
                    UpcomingPayment(name: "iCloud", amount: 2.99, daysLeft: 8)
                ],
                translations: WidgetTranslations(
                    upcoming: "UPCOMING",
                    noPayments: "No upcoming payments",
                    today: "TODAY",
                    dayChar: "d"
                )
            )
        )
    }
    
    /// Empty entry when no data available
    static var empty: SubTrackerEntry {
        SubTrackerEntry(date: Date(), data: nil)
    }
}

File: android/app/src/main/java/com/erayoz/subtracker/widget/WidgetData.kt
package com.erayoz.subtracker.widget

import com.google.gson.annotations.SerializedName

/**
 * Data classes for widget data
 * Matches iOS WidgetData.swift structure
 */

data class WidgetData(
    @SerializedName("currency")
    val currency: String,
    
    @SerializedName("upcomingPayments")
    val upcomingPayments: List<UpcomingPayment>,
    
    @SerializedName("translations")
    val translations: WidgetTranslations,

    @SerializedName("language")
    val language: String?
)

data class UpcomingPayment(
    @SerializedName("name")
    val name: String,
    
    @SerializedName("amount")
    val amount: Double,
    
    @SerializedName("daysLeft")
    val daysLeft: Int
)

data class WidgetTranslations(
    @SerializedName("upcoming")
    val upcoming: String,
    
    @SerializedName("noPayments")
    val noPayments: String,
    
    @SerializedName("today")
    val today: String,
    
    @SerializedName("dayChar")
    val dayChar: String
)
